internal/worker/clienterrors/errors.go

```diff
@@ -0,0 +1,116 @@
+package clienterrors
+
+import (
+	"fmt"
+	"net/http"
+)
+
+const (
+	ErrorCodePrefix = "IMAGE-BUILDER-WORKER-CLIENT-"
+
+	ErrorUnsupportedMediaType ServiceErrorCode = 3
+	ErrorBodyDecodingError    ServiceErrorCode = 4
+	ErrorResourceNotFound     ServiceErrorCode = 5
+	ErrorMethodNotAllowed     ServiceErrorCode = 6
+	ErrorNotAcceptable        ServiceErrorCode = 7
+	ErrorErrorNotFound        ServiceErrorCode = 8
+	ErrorNoDynamicArgs        ServiceErrorCode = 9
+	ErrorInvalidTargetConfig  ServiceErrorCode = 10
+	ErrorSharingTarget        ServiceErrorCode = 11
+	ErrorInvalidTarget        ServiceErrorCode = 12
+
+	// internal errors
+	ErrorDNFError             ServiceErrorCode = 1001
+	ErrorDepsolveJob          ServiceErrorCode = 1002
+	ErrorParsingDynamicArgs   ServiceErrorCode = 1003
+	ErrorManifestGeneration   ServiceErrorCode = 1004
+	ErrorBuildJob             ServiceErrorCode = 1005
+	ErrorUploadingImage       ServiceErrorCode = 1006
+	ErrorRegisteringImage     ServiceErrorCode = 1007
+	ErrorKojiFailedDependency ServiceErrorCode = 1008
+	ErrorKojiBuild            ServiceErrorCode = 1009
+	ErrorInvalidConfig        ServiceErrorCode = 1010
+
+	// Errors contained within this file
+	ErrorUnspecified          ServiceErrorCode = 10000
+	ErrorNotHTTPError         ServiceErrorCode = 10001
+	ErrorServiceErrorNotFound ServiceErrorCode = 10002
+	ErrorMalformedOperationID ServiceErrorCode = 10003
+)
+
+type ServiceErrorCode int
+
+type serviceError struct {
+	code       ServiceErrorCode
+	httpStatus int
+	reason     string
+}
+
+type serviceErrors []serviceError
```

Comment:
If this is a `map[ServiceErrorCode]ServiceError` instead of a slice, we wouldn't need the `find()` function looping through all the errors (do we need it though?)

Reply:
I don't think we do, I just copied what we were doing in other parts of the code. I'll change it to `map[ServiceErrorCode]ServiceError`. I'll get these changes done now

Reply:
Ah yea I implemented the other parts using find(); it's a bit awkward and this code is duplicated a bunch. We could consider unifying some of it in common.

Regardless this slice is just internal to the errors.go file, so if it's easier to use a map then please do.